subsys/bluetooth/mesh/access.c

```diff
@@ -567,6 +613,22 @@ int bt_mesh_comp_data_get_page_1(struct net_buf_simple *buf)
 	comp = bt_mesh_comp_get();
 
 	for (i = 0; i < comp->elem_count; i++) {
+		if (net_buf_simple_tailroom(buf) <
+		    (page1_elem_size(&comp->elem[i]) + BT_MESH_MIC_SHORT)) {
+			if (IS_ENABLED(CONFIG_BT_MESH_LARGE_COMP_DATA_SRV)) {
+				/* Mesh Profile 1.1 Section 4.4.1.2.2:
+				 * If the complete list of models does not fit in the Data field,
+				 * the element shall not be reported.
+				 */
+				LOG_DBG("Element 0x%04x didn't fit in the Data field",
+					comp->elem[i].addr);
+				return 0;
```

Comment:
Is it correct to return ready at this moment? We do not know if we return here at the bottum.

Reply:
The implementation of this check for comp page 0 does the same. Im not 100% sure but I assumed that if `CONFIG_BT_MESH_LARGE_COMP_DATA_SRV` is enabled this is used to send the composition segmented?